gateway/client-switchover/README.md

```diff
@@ -0,0 +1,422 @@
+Please replace this appropriately if the configured domain name is different or if you would prefer to use the loadbalancer IP address instead.
+Also, please change the Kafka bootstrap listener port corresponding to this endpoint appropriately, based on the applied loadbalancer service yaml.
+```
+
+### Step 3: Create source and destination Kafka cluster configuration files.
+
+1. Create source cluster configuration: `source-cluster.config`.
+- Modify the `sasl.jaas.config` section with appropriate credentials and the `bootstrap.servers` section with the appropriate endpoint.
+```
+bootstrap.servers=kafka-source:9093
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="admin" password="admin-secret";
+```
+
+2. Create destination cluster configuration: `destination-cluster.config`.
+- Modify the `sasl.jaas.config` section with appropriate credentials.
+```
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="admin" password="admin-secret";
+```
+
+### Step 4: Test Initial Gateway setup [Loadbalancer pointing to Blue deployment]
+
+1. Test producing messages:
+```
+kafka-console-producer \
+  --bootstrap-server gateway.example.com:9595 \
+  --producer.config source-cluster.config \
+  --topic gateway-blue-test
+```
+2. Test consuming messages:
+```
+kafka-console-consumer \
+  --bootstrap-server gateway.example.com:9595 \
+  --consumer.config source-cluster.config \
+  --topic gateway-blue-test \
+  --from-beginning
+```
+
+### Step 5: Configure Cluster Linking from source to destination Kafka cluster.
+#### NOTE: Please modify the `bootstrap-server` config appropriately for all commands in this section.
+
+#### 1. Create the Cluster Link on the destination cluster
+```
+kafka-cluster-links \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config \
+  --create \
+  --link source-to-destination-link \
+  --config-file source-cluster.config
+```
+
+#### 2. List and Verify Cluster Links.
+- List all cluster links on destination.
+```
+kafka-cluster-links \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config \
+  --list
+```
+
+- Describe the cluster link. 
+```
+kafka-cluster-links \
+  --bootstrap-server kafka-destination:9193 \
+  --command-config destination-cluster.config \
+  --describe \
+  --link source-to-destination-link
+```
+
+### Step 6: Test Cluster linking setup
+#### NOTE: Please modify the `bootstrap-server` config appropriately for all commands in this section.
+
+#### 1. Create Test Topic on Source Kafka Cluster
+```
+bash kafka-topics.sh --create --topic test-topic --bootstrap-server kafka-source:9093  --command-config source-cluster.config
+```
+
+#### 2. Create Mirror Topic on Destination Kafka Cluster
```

Comment:
There is `auto.topic.creation` properties at server end which also client can leverage.